app/services/yelp_service.rb

```diff
@@ -1,24 +1,42 @@
 class YelpService
-  def initialize(activity,latitude, longitude)
-    @activity = activity
+  def initialize(latitude, longitude)
     @latitude = latitude
     @longitude = longitude
   end
 
-  def self.get_restaurants(activity, latitude_value, longitude_value)
-    new(activity, latitude_value, longitude_value).get_restaurants
-  end
-
   def conn
-    Faraday.new("https://api.yelp.com/") do |f|
-      f.headers["Authorization"] = "Bearer #{ENV['YELP_API_KEY']}"
+    Faraday.new("https://api.yelp.com/v3/businesses/search") do |f|
       f.adapter Faraday.default_adapter
+      f.params['latitude'] = @latitude
+      f.params['longitude'] = @longitude
+      f.headers["Authorization"] = "Bearer #{ENV['YELP_API_KEY']}"
     end
   end
 
-  def get_restaurants
-    response = conn.get("v3/businesses/search?term=#{@activity}&latitude=#{@latitude}&longitude=#{@longitude}&limit=25")
-    results = JSON.parse(response.body)["businesses"]
+  def businesses_search(categories, sort_by='distance')
+    search_categories = categories.split(",")
```

Comment:
I really like that we made this work for multiple scenarios and equally-distributed results.